src/main/kotlin/org/rust/lang/core/types/infer/TypeInferenceWalker.kt

```diff
@@ -1461,10 +1451,36 @@ private fun KnownItems.findRangeTy(rangeName: String, indexType: Ty?): Ty {
     return ty.substitute(mapOf(typeParameter to indexType).toTypeSubst())
 }
 
-private fun KnownItems.makeBox(innerTy: Ty): Ty {
-    val box = Box ?: return TyUnknown
-    val boxTy = TyAdt.valueOf(box)
-    return boxTy.substitute(mapOf(boxTy.typeArguments[0] as TyTypeParameter to innerTy).toTypeSubst())
+private fun KnownItems.makeBox(innerTy: Ty): Ty = make("T", innerTy) { Box }
+
+private inline fun <T> KnownItems.make(
+    typeParamName: String,
+    innerTy: Ty,
+    itemSupplier: KnownItems.() -> T?
+): Ty where T : RsTypeDeclarationElement,
+            T : RsGenericDeclaration {
+    val item = itemSupplier() ?: return TyUnknown
```

Comment:
Why we need a function parameter that we call immediately? We could pass `T?` as a parameter.

Reply:
Actually, we don't need it. I've simplified this function
Thanks!